lib/ProvidePlugin.js

```diff
@@ -26,6 +29,9 @@ ProvidePlugin.prototype.apply = function(compiler) {
 			});
 		}
 		compiler.parser.plugin("expression " + name, function(expr) {
+			if(options && !ModuleFilenameHelpers.matchObject(options, this.state.current.resource)) {
```

Comment:
`this.state.module` instead of `this.state.current`, elsewise it doesn't work in `require.ensure` blocks.